File: example/kotlin/rpcimpls/AppServiceImplementations.kt
package rpcimpls

import example.app.BrokenKotlinImpl
import example.app.GameServiceImpl
import rosetta.example.BrokenKotlin
import rosetta.example.GameService
import rosetta.example.GeneratedServiceFactory
import rosetta.example.ServiceImplementations

/**
 * The one hand-written seam the generated Bootstrap expects: an implementation of
 * ServiceImplementations for every service this language implements (GameService and BrokenKotlin).
 * Must live in package `rpcimpls` as `object AppServiceImplementations`, so that Bootstrap.kt can
 * reference it without knowing this project's own package layout.
 */
object AppServiceImplementations : ServiceImplementations {
  override fun gameService(factory: GeneratedServiceFactory): GameService = GameServiceImpl(factory)

  override fun brokenKotlin(factory: GeneratedServiceFactory): BrokenKotlin = BrokenKotlinImpl()
}

File: example/csharp/rpcimpls/AppServiceImplementations.cs
using Example.App;
using Rosetta.Example;

namespace RpcImpls;

/// <summary>
/// The one hand-written seam the generated Bootstrap expects: an
/// implementation of ServiceImplementations for every service this
/// language implements (Achievements and BrokenCSharp). Must live in
/// namespace `RpcImpls` as `AppServiceImplementations`, so that
/// Bootstrap.cs can reference it without knowing this project's own
/// namespace layout.
/// </summary>
public class AppServiceImplementations : ServiceImplementations
{
    public IAchievements? Achievements(GeneratedServiceFactory factory) => new AchievementsServiceImpl(factory);

    public IBrokenCSharp? BrokenCSharp(GeneratedServiceFactory factory) => new BrokenCSharpServiceImpl();
}
